integration/spark/src/main/scala/org/apache/spark/sql/parser/CarbonSpark2SqlParser.scala

```diff
@@ -257,6 +259,25 @@ class CarbonSpark2SqlParser extends CarbonDDLSqlParser {
     (SET ~> "(" ~> repsep(element, ",") <~ ")") ~
     ("=" ~> restInput) <~ opt(";") ^^ {
       case tab ~ columns ~ rest =>
+        // If update is received for complex data types then throw exception
+        var finalColumns = List.empty[String]
+        val updateColumns: java.util.List[String] = new util.ArrayList[String]()
+        columns.foreach { column =>
+          if (column.contains('.')) {
+            val columnFullName = column.split('.')
+            if (columnFullName.length >= 3) {
```

Comment:
you can combine condition 1 and 3 by keeping (columnFullName.length < 3 && condition 2)

Reply:
columnFullName.length >= 3  this case is removed as we can not give normally [tablename.complextype.childtype].